### ISSUE TITLE
Downloaded indexed images lost on upload

### ISSUE THREAD

**Issue by outer-core** [chilli-axe/mpc-autofill#113]
I'm having issues with an order that has two (or more) images with the same name once it's uploaded to MPC.

When the images are downloaded, Chrome will rename them to add a unique index to avoid a conflict, however the uploader doesn't seem to keep track of the renamed image(s) and so once uploaded to MPC only the first (non-indexed) image is used for all slots.

You should hopefully be able to reproduce with this order (or any similar type of order):

```
<order><details><quantity>2</quantity><bracket>18</bracket><stock>(S30) Standard Smooth</stock><foil>false</foil></details><fronts><card><id>1v9I-3jRXthjkbDxI0wBY4-3pkuKaoGNE</id><slots>1</slots><name>Watery Grave.png</name><query>watery grave</query></card><card><id>10PYkBtUG_PNW5Hmu5aB5FJ5TkzxBo0x2</id><slots>0</slots><name>Watery Grave.png</name><query>watery grave</query></card></fronts><cardback>1LrVX0pUcye9n_0RtaDNVl2xPrQgn7CYf</cardback></order>
```

MPCFill:
![Screenshot (1)](https://user-images.githubusercontent.com/104941914/204116985-d455e090-1941-493e-861e-0b1eeedada25.png)

MPC:
![Screenshot (2)](https://user-images.githubusercontent.com/104941914/204116992-1f6eeb6e-26b1-408f-aa8d-6aaf2c139911.png)

_Thanks for a great tool and workflow, it's awesome!_

**Comment by ndepaola** [comment]
it sounds like you:
* downloaded the full-res images through the site (e.g. with the download all button)
* generated the xml for your project
* put the xml into a folder
* created a `cards` folder next to the xml
* put your downloaded images into the `cards` folder
* ran the tool

is that correct?

the tool is set up to automatically download the images required for your project and store them in the `cards` folder. when it does this, it modifies the filename to contain the file's google drive ID in parentheses - e.g. `Watery Grave (google drive ID).png`. this prevents collisions when a project contains two cards with the same name.

however, when checking if an image needs to be downloaded, the tool will also see if a file exists with the card's name *without google drive id in parentheses*. if this file exists, the tool will use it rather than downloading the image again from gdrive. the tool has no way of differentiating which card should use `Watery Grave.png` and which one should use `Watery Grave (1).png`.

so i have two notes:
a) can you clear the `cards` folder and let the tool download the images for you (as this will ensure the naming collision is handled)?
b) iirc it's not possible to change the names of files downloaded through the web interface for google drive, but i can look into this some more. ideally the frontend should also handle putting the filename in parentheses to avoid this issue.

**Comment by outer-core** [comment]
> is that correct?

That is indeed what I did. And your directions do fix my issue. Upon review, I see my workflow was counter the design (and not what was documented). So you can consider this user issue (sorry about that) closed.

---

Documentation aside, I suppose I didn't realize the `Download All Images` button was a utility feature and not part of the overall workflow.

Thinking out loud, I wonder if it would've been more clear for a new user (e.g. me) if the button were instead like:

```
+-------------------------------+      +------------------------------------+
| Generate Order | What's Next? |      |           Generate Order           |
+----------------+--------------+  or  +--------------+---------------------+
| Download All Images (Utility) |      | What's Next? | Download All Images |
+-------------------------------+      |              |     (Utility)       |
                                       +--------------+---------------------+
```

Where `Generate Order` was green in color (i.e. 'done') and the others are a separate/standalone color. But of course, that would break existing user habits so that change is likely unwarranted.

> b) iirc it's not possible to change the names of files downloaded through the web interface for google drive, but i can look into this some more. ideally the frontend should also handle putting the filename in parentheses to avoid this issue.

Shame that's not possible - it would enable additional user workflows to more easily tweak the images, e.g. : 1) download all images, 2) generate order, 3) user makes manual edits to images as desired, 4) upload to MPC via `.exe`. Instead of the existing: 1) generate order, 2) download and upload to MPC via `.exe`.

**Comment by ndepaola** [comment]
understood, I can see how the intended workflow isn't totally clear from the editor's UI. i'll leave this issue open until i address one or both of:
* communicating in the editor UI that downloading images manually isn't typically required as the tool will do it for you
* inserting the file's gdrive ID in parentheses when downloading images through the frontend (if possible)

**Comment by ndepaola** [comment]
this item is fixed in the new frontend: https://github.com/chilli-axe/mpc-autofill/pull/149 i'll close this issue once the new frontend is ready for public use

**Comment by ndepaola** [comment]
the public beta is available now - you can access it with the mpcfill.com database at https://mpcautofill.github.io?server=https://mpcfill.com - where this issue is fixed.
